sdk/identity/identity/samples/AzureCacheForRedis/ioredis.md

```diff
@@ -0,0 +1,273 @@
+## Azure Cache for Redis: Azure AD with ioredis client library
+
+### Table of contents
+
+- [Prerequisites](#prerequisites)
+- [Authenticate with Azure AD - Hello World](#authenticate-with-azure-ad-ioredis-hello-world)
+- [Authenticate with Azure AD - Handle Reauthentication](#authenticate-with-azure-ad-handle-reauthentication)
+- [Authenticate with Azure AD - Using Token Cache](#authenticate-with-azure-ad-using-token-cache)
+- [Troubleshooting](#troubleshooting)
+
+#### Prerequisites
+
+- Configuration of Role and Role Assignments is required before using the sample code in this document.
+- **Dependency Requirements:**
+    Add the following dependencies to *package.json*:
+
+    ```
+    "dependencies": {
+      "@azure/identity": "^2.0.4",
+      "ioredis": "^5.0.4",
+    ```
+- Familiarity with the [ioredis](https://github.com/luin/ioredis) and [Azure Identity for JavaScript](https://docs.microsoft.com/javascript/api/overview/azure/identity-readme?view=azure-node-latest) client libraries is assumed.
+
+#### Samples Guidance
+
+* [Authenticate with Azure AD - Hello World](#authenticate-with-azure-ad-hello-world):
+   This sample is recommended for users getting started to use Azure AD authentication with Azure Cache for Redis.
+* [Authenticate with Azure AD - Handle Reauthentication](#authenticate-with-azure-ad-handle-reauthentication):
+   This sample is recommended to users looking to build long-running applications and would like to handle reauthenticating with Azure AD upon token expiry.
+* [Authenticate with Azure AD - Using Token Cache](#authenticate-with-azure-ad-using-token-cache):
+  This sample is recommended to users looking to build long-running applications that would like to handle reauthenticating with a token cache. The token cache stores and proactively refreshes the Azure AD access token 2 minutes before expiry and ensures a non-expired token is available for use when the cache is accessed.
+
+#### Authenticate with Azure AD: Hello World
+
+This sample is intended to assist in authenticating a hosted Azure Cache for Redis instance with Azure AD via the ioredis client library. It focuses on displaying the logic required to fetch an Azure AD access token and to use it as the password when setting up the ioredis instance.
+
+##### Migration Guidance
+
+When migrating your existing application code to authenticate with Azure AD, replace the password input with Azure AD token. Azure Redis Cache name, username, Azure AD token, and use of SSL are required while connecting with the cache.
+The username will depend on whether you're using service principal, managed identity, or Microsoft username.
```

Comment:
I feel that this should be the leading bit of file. It's the single most important thing the sample conveys: to authenticate with Azure Cache for Redis using AAD RBAC, replace the username with the name of an app registration with the correct role binding, and pass a valid access token to the password field.